src/performance/network.ts

```diff
@@ -0,0 +1,46 @@
+import { Event } from "@clarity-types/data";
+import { NetworkState } from "@clarity-types/performance";
+import config from "@src/core/config";
+import { schedule } from "@src/core/task";
+import { clearTimeout, setTimeout } from "@src/core/timeout";
+import encode from "./encode";
+
+export let state: NetworkState[] = [];
+let timeout: number = null;
+
+export function compute(entry: PerformanceResourceTiming): void {
```

Comment:
>PerformanceResourceTiming [](start = 31, length = 25)

Small preference: Keep the resource name of the fail and event type from before. That keeps the naming alignment between "xxx Timing" and xxx consistent.

Reply:
In this case PerformanceResourceTiming is a native Typescript type from W3 specs. We will have to rename the type and redefine custom type. I will avoid adding custom types where possible. Please let me know if you think I missed something here.

---
In reply to: [347585636](https://github.com/microsoft/clarity-js/pull/160#discussion_r347585636) [](ancestors = 347585636)